postmortems/002 - Postmortem for v0.0.23 release.md

```diff
@@ -0,0 +1,31 @@
+# 002 - Postmortem for v0.0.23 release
+
+## Level: Internal
+
+## Author(s)
+
+- David Ellis <david@iasql.com>
+
+## Summary
+
+We have been unable to release v0.0.23, a significant rearchitecting and feature-adding version of IaSQL, for approximately a month, *primarily* due to issues with the new infrastructure required in staging and production. There are five incidences we have observed on staging (none on production as production is not yet in use) and we have applied mitigations to staging and production for them all. This document will outline them.
+
+## Timeline
+
+- **2022-12-20**: Staging cannot load docker container images because it has run out of disk space. Increased boot volume to 16GB.
+- **2022-12-27**: Staging deploys fail complaining about lack of disk space, again. Dropping `stdout/err` in favor of LogDNA recommended.
+- **2023-01-03**: Staging `run` service crashed, cannot be deployed automatically. Deployment code written, better logging and auto-restart added to `run` service. Discord altering on staging/production status revived.
+- **2023-01-08**: Staging instance becomes unreachable. Somehow the DHCP daemon failed to get an IP address assigned by AWS. Manually stopping and starting the server revived it.
+- **2023-01-11**: Staging cannot deploy. Authorization to ghcr.io failed. Replaced Github Personal Access Token. Recommend monthly calendar task to replace the token until ghcr.io is replaced with dockerhub (potentially).
+
+## Detection
+
+Our test suite for the Dashboard uses staging, so tests there start failing, but with low activity over the holidays, this was delayed for the January 3rd situation. The first and last issue happened during deployment so they were noticed immediately. The lack of disk space on December 27th and the instance being unreachable on January 8th were only spotted by deployment failures (but not caused by them). They should be spotted sooner now that Discord health notifications have been revived.
```

Comment:
In the case of an instance lack of disk space, does the instance becomes unhealthy or should we try to create another check to the instance so we can get notified through discord?

Reply:
Running out of space on the boot disk immediately causes chaos with Docker and a couple of other services on the instance, so we *should* be alerted.